### DISCUSSION TITLE
Allow `tests` nested within a `test`

### DISCUSSION THREAD

**Discussion by DrHyde** [ctrf-io/ctrf#14] [Schema]
I'm primarily a perl dev, but came across CTRF when one of my more JS-ish colleagues used it. Obviously having only one format for test reports in a multi-language project is a Good Thing, so I thought I'd look at writing a formatter along the lines of [TAP::Formatter::JUnit](https://metacpan.org/pod/TAP::Formatter::JUnit) to spit out CTRF. Alas, it isn't possible, because perl's standard testing tools and [TAP](https://testanything.org/) support arbitrarily deeply nested "subtests":

```
$ cat foo.t
use Test::More;

pass("this isn't a subtest");
subtest "here is a subtest" => sub {
    pass("a pass in a subtest");
    fail("a fail in a subtest");
    subtest "nested subtest" => sub {
        fail("a fail in a sub-subtest");
    };
};
pass("this also isn't a subtest");
done_testing;

$ prove -v foo.t
foo.t ..
ok 1 - this isn't a subtest
# Subtest: here is a subtest
    ok 1 - a pass in a subtest
    not ok 2 - a fail in a subtest
    # Subtest: nested subtest
        not ok 1 - a fail in a sub-subtest
        1..1
    not ok 3 - nested subtest
    1..3
not ok 2 - here is a subtest
ok 3 - this also isn't a subtest
1..3
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests

Test Summary Report
-------------------
foo.t (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
Files=1, Tests=3,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.04 cusr  0.01 csys =  0.08 CPU)
Result: FAIL
```

If you extend the schema to allow a `tests` object in a `test` then I could do it.

**Comment by timkrins** [comment]
In Ruby, rspec also has this concept (although named differently [contexts]).

Each subtest is just a test.

I believe in most rspec test reporting outputs, the subtest name is prefixed with the test/context name.

Could you flatten your output in such a way to fit CTRF?
